jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -182,15 +175,14 @@ public void onTimeout(AsyncEvent event) throws IOException
         @Override
         public void onError(AsyncEvent event) throws IOException
         {
-            complete(((HttpServletRequest)event.getAsyncContext().getRequest()).getSession(false));
+            complete(Request.getBaseRequest(event.getAsyncContext().getRequest()).getSession(false));
```

Comment:
What happens if `Request.getBaseRequest(HttpServletRequest)` returns null? (which it can)

Reply:
You're not wrong, however there's 26 other instances in the code where Request.getBaseRequest is called and null is never checked. So, this can be instance 27, and when/if we need to fix we can fix all of the usages in one go.